File: Fifo-Match-Application-main/Fifo-Match-Application-main/app/src/main/java/com/fifo/match/model/UserDetailsModel.kt
package com.fifo.match.model


import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class UserDetailsModel(
    @SerializedName("user_details")
    val userDetails: UserDetails?
):Serializable {
    data class UserDetails(
        @SerializedName("id")
        val id: Int?,
        @SerializedName("firebase_id")
        val firebaseId: String?,
        @SerializedName("social_type")
        val socialType: String?,
        @SerializedName("social_id")
        val socialId: String?,
        @SerializedName("name")
        val name: String?,
        @SerializedName("email")
        val email: String?,
        @SerializedName("is_email_verified")
        val isEmailVerified: Boolean?,
        @SerializedName("mobile")
        val mobile: String?,
        @SerializedName("interested_in")
        val interestedIn: String?,
        @SerializedName("gender")
        val gender: String?,
        @SerializedName("profile_pic")
        val profilePic: Any?,
        @SerializedName("dob")
        val dob: String?,
        @SerializedName("about")
        val about: String?,
        @SerializedName("status")
        val status: Int?,
        @SerializedName("subscription_plan")
        val subscriptionPlan: Boolean?,
        @SerializedName("device_type")
        val deviceType: String?,
        @SerializedName("app_version")
        val appVersion: String?,
        @SerializedName("os_version")
        val osVersion: String?,
        @SerializedName("auth_token")
        val authToken: String?,
        @SerializedName("latitude")
        val latitude: String?,
        @SerializedName("longitude")
        val longitude: String?,
        @SerializedName("current_latitude")
        val currentLatitude: String?,
        @SerializedName("current_longitude")
        val currentLongitude: String?,
        @SerializedName("last_activity")
        val lastActivity: Any?,
        @SerializedName("verify")
        val isVerified: Int?,
        @SerializedName("relationship_status_id")
        val relationshipStatusId: Int?,
        @SerializedName("occupation_id")
        val occupationId: Int?,
        @SerializedName("education_id")
        val educationId: Int?,
        @SerializedName("is_pause")
        val isPause: Boolean?,
        @SerializedName("flowers")
        val flowers: Int?,
        @SerializedName("profile_complete")
        val profileComplete: Int?,
        @SerializedName("created_at")
        val createdAt: String?,
        @SerializedName("updated_at")
        val updatedAt: String?,
        @SerializedName("deleted_at")
        val deletedAt: Any?,
        @SerializedName("working_fifo")
        val workingFifo: String?,
        @SerializedName("swing")
        val swing: String?,
        @SerializedName("min_age")
        val minAge: Any?,
        @SerializedName("max_age")
        val maxAge: Any?,
        @SerializedName("login_time")
        val loginTime: String?,
        @SerializedName("country_code")
        val countryCode: Any?,
        @SerializedName("country_name")
        val countryName: String?,
        @SerializedName("surey_status")
        val sureyStatus: Int?,
        @SerializedName("miles")
        val miles: String?,
        @SerializedName("age")
        val age: Int?,
        @SerializedName("is_blocked")
        val isBlocked: Boolean?,
        @SerializedName("is_like")
        val isLike: Boolean?,
        @SerializedName("is_chat")
        val isChat: Boolean?,
        @SerializedName("is_saved")
        val isSaved: Boolean?,
        @SerializedName("photos")
        val photos: List<Photo>,
        @SerializedName("questionnaire")
        val questionnaire: Questionnaire?,
        @SerializedName("occupation")
        val occupation: Occupation?,
        @SerializedName("relationship_data")
        val relationshipData: RelationshipData?,
        @SerializedName("education")
        val education: Education?,
        @SerializedName("user_inapp_transaction")
        val userInappTransaction: UserInappTransaction?
    ) {
        data class Photo(
            @SerializedName("id")
            val id: Int?,
            @SerializedName("user_id")
            val userId: Int?,
            @SerializedName("type")
            val type: Int?,
            @SerializedName("name")
            val name: String?,
            @SerializedName("created_at")
            val createdAt: String?,
            @SerializedName("updated_at")
            val updatedAt: String?,
            @SerializedName("deleted_at")
            val deletedAt: Any?
        )

        data class Questionnaire(
            @SerializedName("id")
            val id: Int?,
            @SerializedName("user_id")
            val userId: Int?,
            @SerializedName("height_type")
            val heightType: String?,
            @SerializedName("height")
            val height: String?,
            @SerializedName("body_type")
            val bodyType: String?,
            @SerializedName("seeking")
            val seeking: String?,
            @SerializedName("qualities_appreciate")
            val qualitiesAppreciate: String?,
            @SerializedName("personality_types")
            val personalityTypes: String?,
            @SerializedName("good_looking")
            val goodLooking: String?,
            @SerializedName("kids")
            val kids: String?,
            @SerializedName("kids_in_future")
            val kidsInFuture: String?,
            @SerializedName("my_health")
            val myHealth: String?,
            @SerializedName("other_health")
            val otherHealth: String?,
            @SerializedName("relevant")
            val relevant: Any?,
            @SerializedName("my_qualities")
            val myQualities: String?,
            @SerializedName("other_qualities")
            val otherQualities: Any?,
            @SerializedName("personality")
            val personality: String?,
            @SerializedName("interested_fact")
            val interestedFact: String?,
            @SerializedName("decision")
            val decision: String?,
            @SerializedName("prefer_clarity")
            val preferClarity: String?,
            @SerializedName("created_at")
            val createdAt: String?,
            @SerializedName("updated_at")
            val updatedAt: String?,
            @SerializedName("deleted_at")
            val deletedAt: Any?
        )

        data class Occupation(
            @SerializedName("id")
            val id: Int?,
            @SerializedName("name")
            val name: String?,
            @SerializedName("status")
            val status: String?,
            @SerializedName("created_at")
            val createdAt: String?
        )

        data class RelationshipData(
            @SerializedName("id")
            val id: Int?,
            @SerializedName("name")
            val name: String?,
            @SerializedName("status")
            val status: String?,
            @SerializedName("created_at")
            val createdAt: String?
        )

        data class Education(
            @SerializedName("id")
            val id: Int?,
            @SerializedName("name")
            val name: String?,
            @SerializedName("status")
            val status: String?,
            @SerializedName("created_at")
            val createdAt: String?
        )
        data class UserInappTransaction(
            @SerializedName("transaction_id")
            val transactionId: String?,
            @SerializedName("user_id")
            val userId: Int?,
            @SerializedName("type")
            val type: String?,
            @SerializedName("plan_id")
            val planId: String?,
            @SerializedName("membership_type")
            val membershipType: String?,
            @SerializedName("membership_type_value")
            val membershipTypeValue: Int?,
            @SerializedName("amount")
            val amount: String?,
            @SerializedName("start_datetime")
            val startDatetime: String?,
            @SerializedName("end_datetime")
            val endDatetime: String?,
            @SerializedName("status")
            val status: Boolean?,
            @SerializedName("created_at")
            val createdAt: String?,
            @SerializedName("updated_at")
            val updatedAt: String?
        )
    }
}

File: Fifo-Match-IOS-main/Fifo-Match-IOS-main/FIFOMatch/Models/Home/UserDetailsModel.swift
//
//  UserDetailsModel.swift
//  FIFOMatch
//
//  Created by Harendra Singh Rathore on 30/03/22.
//

import Foundation

// MARK: - Welcome
struct UserDetailsModel: Decodable {
  let msg, message: String?
  let status: Int?
  let data: UserData?
}

// MARK: - DataClass
struct UserData: Decodable {
  let userDetails: UserDetails?
  
  enum CodingKeys: String, CodingKey {
    case userDetails = "user_details"
  }
}

// MARK: - UserDetails
struct UserDetails: Decodable {
  let id: Int?
  let firebaseID, socialType, socialID, name: String?
  let email: String?
  let isEmailVerified: Bool?
  let mobile, interestedIn, gender, profilePic: String?
  let dob, about: String?
  let status: Int?
  let subscriptionPlan: Bool?
  let deviceType, appVersion, osVersion, authToken: String?
  let latitude, longitude, currentLatitude, currentLongitude: String?
  let isVerified: Bool?
  let relationshipStatusID, occupationID, educationID: Int?
  let isPause: Bool?
  let flowers, profileComplete: Int?
  let workingFIFO, swing: String?
  let minAge, maxAge: Int?
  let loginTime, countryCode, countryName: String?
  let sureyStatus: Int?
  let miles: String?
  let age: Int?
  let photos: [UserPhoto]?
  let questionnaire: SurveyDetailData?
  let occupation, relationshipData, education: Education?
  let userInappTransaction: UserInappTransaction?
  let isBlocked: Bool?
  let isLike: Bool?
  let isSaved: Bool?
  let verify: Int?
  let isChatEnable: Bool?
  
  enum CodingKeys: String, CodingKey {
    case id
    case firebaseID = "firebase_id"
    case socialType = "social_type"
    case socialID = "social_id"
    case name, email
    case isEmailVerified = "is_email_verified"
    case mobile
    case interestedIn = "interested_in"
    case gender
    case profilePic = "profile_pic"
    case dob, about, status
    case subscriptionPlan = "subscription_plan"
    case deviceType = "device_type"
    case appVersion = "app_version"
    case osVersion = "os_version"
    case authToken = "auth_token"
    case latitude, longitude
    case currentLatitude = "current_latitude"
    case currentLongitude = "current_longitude"
    case isVerified = "is_verified"
    case relationshipStatusID = "relationship_status_id"
    case occupationID = "occupation_id"
    case educationID = "education_id"
    case isPause = "is_pause"
    case flowers
    case profileComplete = "profile_complete"
    case workingFIFO = "working_fifo"
    case swing = "swing"
    case minAge = "min_age"
    case maxAge = "max_age"
    case loginTime = "login_time"
    case countryCode = "country_code"
    case countryName = "country_name"
    case sureyStatus = "surey_status"
    case miles, age, photos, questionnaire, occupation
    case relationshipData = "relationship_data"
    case education
    case userInappTransaction = "user_inapp_transaction"
    case isBlocked = "is_blocked"
    case isLike = "is_like"
    case isSaved = "is_saved"
    case verify
    case isChatEnable = "is_chat"
  }
}

// MARK: - Education
struct Education: Decodable {
  let id: Int?
  let name, status: String?
}

// MARK: - UserInappTransaction
struct UserInappTransaction: Decodable {
    let transactionID: String?
    let userID: Int?
    let type, planID, membershipType: String?
    let membershipTypeValue: Int?
    let amount, startDatetime, endDatetime: String?
    let status: Bool?
    let createdAt, updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case transactionID = "transaction_id"
        case userID = "user_id"
        case type
        case planID = "plan_id"
        case membershipType = "membership_type"
        case membershipTypeValue = "membership_type_value"
        case amount
        case startDatetime = "start_datetime"
        case endDatetime = "end_datetime"
        case status
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}
